actionview/lib/action_view/template.rb

```diff
@@ -101,6 +101,54 @@ class Template
     # You can use +local_assigns+ in the sub templates to access the local variables:
     #
     #   local_assigns[:headline] # => "Welcome"
+    #
+    # Each key in +local_assigns+ is available as a partial-local variable:
+    #
+    #   local_assigns[:headline] # => "Welcome"
+    #   headline                 # => "Welcome"
+    #
+    # Since +local_assigns+ is a +Hash+, it's compatible with Ruby 3.1's pattern
+    # matching assignment operator:
+    #
+    #   local_assigns => { headline:, **options }
+    #   headline                 # => "Welcome"
+    #   options                  # => {}
+    #
+    # Pattern matching assignment also supports variable renaming:
+    #
+    #   local_assigns => { headline: title }
+    #   headline                 # => "Welcome"
```

Comment:
I'm not that familiar with Ruby's pattern matching.
Are you sure this assigns `headline`?

```ruby
local_assigns = { headline: "Welcome" }
local_assigns => { headline: title }
headline # raises undefined local variable or method `headline'
title # => "Welcome"
```

Reply:
In the case of an Action View template, the `headline` variable is assigned as part of template's [compile](https://github.com/rails/rails/blob/d21d811ffece4d3959bcd37e58fec77590ff6f93/actionview/lib/action_view/template.rb#L377-L383) method, namely the [locals_code](https://github.com/rails/rails/blob/d21d811ffece4d3959bcd37e58fec77590ff6f93/actionview/lib/action_view/template.rb#L466-L477) portion.

That means that there's _already_ a `headline` variable before the `<% local_assigns => ... %>` code is executed.

Is that confusing or interesting enough to warrant further explanation in these changes?

Reply:
To me the `headline` line was confusing because to me it suggest `headline` gets assigned by the pattern match, just like the examples above it.

Reply:
I've removed that line. Thank you!